From: Marcin Kulisz <debian@kulisz.net>
Date: Thu, 3 Jun 2021 20:10:24 +0100
Subject: Added explicit string casing (Closes: #986647)

---
 s3 | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/s3 b/s3
index 01b1d9d..83a6dfa 100755
--- a/s3
+++ b/s3
@@ -187,8 +187,9 @@ class AWSCredentials(object):
 
         if data.get("AccessKeyId") is None:
             self.__get_role()
-            data = self.__request_json(urllib.parse.urljoin(self.credentials_metadata,
-                                                            self.iamrole))
+            data = self.__request_json(urllib.parse.urljoin(
+                str(self.credentials_metadata, 'utf-8'),
+                str(self.iamrole, 'utf-8')))
 
         self.access_key = data['AccessKeyId']
         if self.access_key is None or self.access_key == '':
